### PR TITLE
e2e: allow disabling tls certificate validation

### PR DESCRIPTION
**Pull request by sadlerap**
Supercedes #124.

### CODE REVIEW

File: e2e/hook/config.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import (
+	"os"
+
+	"k8s.io/client-go/rest"
+)
+
+const envVarUseInsecure string = "E2E_USE_INSECURE_TLS"
+
+func setTLSFromEnv(cfg *rest.Config) {
+	cfg.Insecure = os.Getenv(envVarUseInsecure) != "false"
```
Comment (filariow) [inline e2e/hook/config.go:12]: IIRC, if cfg.Insecure is set to true and `CAFile` or `CAData` are also set, we'll get an error

File: e2e/hook/config.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package hook
```
Comment (filariow) [inline e2e/hook/config.go:1]: maybe we can move this in an internal package (like `hook/internal/config`)? WDYT?

File: e2e/hook/config.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	cfg.Insecure = os.Getenv(envVarUseInsecure) != "false"
+}
+
+type configMutationFunc func(*rest.Config)
+
+var mutationFuncs = []configMutationFunc{
+	setTLSFromEnv,
+}
+
+func ProcessConfig(cfg *rest.Config) {
```
Comment (filariow) [inline e2e/hook/config.go:21]: I'd say `MutateConfig` gives a better idea of what kind of processing it's doing. WDYT?

```suggestion
func MutateConfig(cfg *rest.Config) {
```

Comment (filariow) [inline e2e/hook/config.go:21]: Having some comments here would be great

**Review by filariow — CHANGES_REQUESTED** [review]
mostly LGTM, some ideas and nits

File: e2e/hook/config.go (hunk 1 above)
Comment (sadlerap) [reply e2e/hook/config.go:12]: I don't see that reflected in the docs for [TLSClientConfig](https://pkg.go.dev/k8s.io/client-go@v0.30.0/rest#TLSClientConfig).  Do you have a source for that?

Comment (filariow) [reply e2e/hook/config.go:12]: PTAL: https://gist.github.com/filariow/631546cba9ea57fe93e605473be78bf5

File: e2e/hook/config.go (hunk 2 above)
Comment (sadlerap) [reply e2e/hook/config.go]: Done in 9a83a18.

File: e2e/hook/config.go (hunk 3 above)
Comment (sadlerap) [reply e2e/hook/config.go]: Done in 9a83a18.

Comment (sadlerap) [reply e2e/hook/config.go]: Fixed in 9a83a18.

File: e2e/hook/config.go (hunk 1 above)
Comment (sadlerap) [reply e2e/hook/config.go]: Fixed in c72a3ce.

File: e2e/hook/internal/config.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"os"
+
+	"k8s.io/client-go/rest"
+)
+
+const envVarUseInsecure string = "E2E_USE_INSECURE_TLS"
+
+func setTLSFromEnv(cfg *rest.Config) {
+	if os.Getenv(envVarUseInsecure) != "false" {
```
Comment (filariow) [inline e2e/hook/internal/config.go:12]: I'd suggest to change this to checking for explicit request of the feature

```suggestion
	if os.Getenv(envVarUseInsecure) == "true" {
```

**Review by filariow — CHANGES_REQUESTED** [review]
just a nit, other than that LGTM

**Pull request merged**